tests/Unit/API/Site/Controllers/CountryCodeTraitTest.php

```diff
@@ -0,0 +1,248 @@
+		protected $iso_provider;
+
+		/** @var MockObject|GoogleHelper $google_helper */
+		protected $google_helper;
+
+		/** @var bool $country_supported */
+		protected $country_supported;
+
+		/**
+		 * Runs before each test is executed.
+		 */
+		public function setUp(): void {
+			parent::setUp();
+
+			$this->iso_provider  = $this->createMock( ISO3166DataProvider::class );
+			$this->google_helper = $this->createMock( GoogleHelper::class );
+
+			// phpcs:ignore WordPress.Classes.ClassInstantiation.MissingParenthesis
+			$this->trait = new class {
+				use CountryCodeTrait {
+					get_country_code_sanitize_callback as public;
+					get_country_code_validate_callback as public;
+					get_supported_country_code_validate_callback as public;
+				}
+			};
+
+			$this->trait->set_iso3166_provider( $this->iso_provider );
+			$this->trait->set_google_helper_object( $this->google_helper );
+
+			$this->country_supported = true;
+
+			$this->google_helper->method( 'is_country_supported' )
+				->willReturnCallback(
+					function () {
+						return $this->country_supported;
+					}
+				);
+		}
+
+		/**
+		 * Runs after each test is executed.
+		 */
+		public function tearDown(): void {
+			self::$rest_validate_request_arg_callback = null;
+		}
+
+		/**
+		 * Test all get_*_callback methods return a callable.
+		 */
+		public function test_all_get_callbacks_return_callable() {
+			$this->assertIsCallable( $this->trait->get_country_code_sanitize_callback() );
+			$this->assertIsCallable( $this->trait->get_country_code_validate_callback() );
+			$this->assertIsCallable( $this->trait->get_supported_country_code_validate_callback() );
+		}
+
+		/**
+		 * Test the get_country_code_validate_callback and get_supported_country_code_validate_callback
+		 * methods will apply arguments to rest_validate_request_arg_callback and return early if invalid.
+		 */
+		public function test_all_get_validate_callbacks_with_rest_validate_request_arg() {
+			$called_count = 0;
+
+			self::$rest_validate_request_arg_callback = function () use ( &$called_count ) {
+				// When it returns a value other than `true`, it means the validation did not pass,
+				// and here we also make it return a called count.
+				$called_count++;
+				return [ $called_count, func_get_args() ];
+			};
+
+			$args = [ 'US', new Request(), 'country_code' ];
+
+			$callback = $this->trait->get_country_code_validate_callback();
+			$results  = $callback( ...$args );
+
+			$this->assertEquals( [ 1, $args ], $results );
+
+			$callback = $this->trait->get_supported_country_code_validate_callback();
+			$results  = $callback( ...$args );
+
+			$this->assertEquals( [ 2, $args ], $results );
+		}
+
+		/**
+		 * Test a sanitized country code with a string.
+		 */
+		public function test_get_country_code_sanitize_callback_with_string() {
+			$callback = $this->trait->get_country_code_sanitize_callback();
+			$result   = $callback( 'us' );
+
+			$this->assertEquals( 'US', $result );
```

Comment:
Instead of setting `$callback` and `$result` for each test we could reduce that to a single line (_where possible..._). It looks like this would be applicable to quite a few of the tests and make the file a bit shorter and tidier.

```suggestion
			$this->assertEquals( 'US', $this->trait->get_country_code_sanitize_callback()( 'us' ) );
```

Reply:
I only apply the suggestion to this test case. If the rest test cases were applied, a single line would be too complicated and even might have to scroll the editor to see the whole line of code, which reduces the readability too much.